dashboard/modules/reporter/reporter_agent.py

```diff
@@ -589,9 +589,11 @@ async def _perform_iteration(self, publisher):
         """Get any changes to the log files and push updates to kv."""
         while True:
             try:
-                formatted_status_string = internal_kv._internal_kv_get(
-                    DEBUG_AUTOSCALING_STATUS
-                )
+                formatted_status_string = None
```

Comment:
Don't we also need to put other logic related to formatted_status_string to this block? Also, please add a comment why it is only for a head node. 

```python3
stats = self._get_all_stats()
formatted_status_string = None
# Cluster stats don't need to be recorded on all agents, so we only report it in a head node. 
if self._is_head_node:
    formatted_status_string = internal_kv._internal_kv_get(DEBUG_AUTOSCALING_STATUS)
    cluster_stats = (
        json.loads(formatted_status_string.decode())
            if formatted_status_string else {})
    records_reported = self._record_stats(stats, cluster_stats)
    self._metrics_agent.record_reporter_stats(records_reported)
await publisher.publish_resource_usage(self._key, jsonify_asdict(stats))
```

Reply:
No, the logic is quite convoluted. Basically, metrics agent will only log AUTOSCALING_STATUS on head node agent, but log other stats on non-head-node.

I think I will make it less confusing